pkg/controller/core/workload_controller.go

```diff
@@ -419,7 +435,8 @@ func (r *WorkloadReconciler) reconcileMaxExecutionTime(ctx context.Context, wl *
 	if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadDeactivationTarget) {
 		workload.SetDeactivationTarget(wl, kueue.WorkloadMaximumExecutionTimeExceeded, "exceeding the maximum execution time")
 		wl.Status.AccumulatedPastExexcutionTimeSeconds = nil
-		if err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true, r.clock); err != nil {
+		err := workload.ApplyAdmissionStatus(ctx, r.client, wl, true, r.clock)
+		if client.IgnoreNotFound(err) != nil {
```

Comment:
Please revert this change and check the usage of reconcileMaxExecutionTime in the Reconcile function:

https://github.com/kubernetes-sigs/kueue/blob/9c4ab1a87fbb1889247c4a4e92d231bee6732e31/pkg/controller/core/workload_controller.go#L364-L367

With the current implementation, reconciliation proceeds to the next steps, but we actually need to stop it on the error.